src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -188,10 +198,32 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
 
   List<SourceRecord> sourceRecords(ConsumerRecords<byte[], byte[]> pollResult) {
     List<SourceRecord> sourceRecords = new ArrayList<>(pollResult.count());
-    pollResult.forEach(sourceRecord -> sourceRecords.add(toSourceRecord(sourceRecord)));
+    pollResult.forEach(
+        consumerRecord -> {
+          if (replayPolicy == ReplayPolicy.FILTER && !isSkippedRecord(consumerRecord)) {
```

Comment:
Might give us back some performance gain although it is done again within the method `isSkippedRecord`

Reply:
I'd put it in one place or the other, not both. I doubt the performance difference is significant.